core/market/decentralized/src/matcher/matcher.rs

```diff
@@ -46,8 +47,14 @@ pub enum MatcherError {
     DemandError(#[from] DemandError),
     #[error(transparent)]
     OfferError(#[from] OfferError),
-    #[error("Internal error: {0}.")]
-    InternalError(String),
+    #[error(transparent)]
+    ErrorMessage(#[from] ErrorMessage),
```

Comment:
ErrorMessage is not meaningful name. InternalError was better

Reply:
both `ErrorMessage` and `DbError` to be converted and stored under `UnexpectedError`